iavl/iavl_proof.go

```diff
@@ -17,20 +18,28 @@ type IAVLProof struct {
 	RootHash   []byte
 }
 
-func (proof *IAVLProof) Verify(key []byte, value []byte, root []byte) bool {
+func (proof *IAVLProof) Verify(key []byte, value []byte, root []byte, version int) bool {
```

Comment:
Yes, this seems wrong to me too. It is breaking the encapsulation. I had to expose it because of the existing tests, but I was thinking of just redoing them at some point. Also, we expose Hash() and HashWithCount() for testing, but I think they should be buried underneath the Save.